src/openeo_gfmap/fetching/commons.py

```diff
@@ -76,19 +79,53 @@ def filter_condition(band_name, _):
     )
 
 
+def _load_collection_hybrid(
+    connection: openeo.Connection,
+    is_stac: bool,
+    collection_id_or_url: str,
+    bands: list,
+    spatial_extent: Union[Dict[str, float], Parameter, None] = None,
+    temporal_extent: Union[Sequence[InputDate], Parameter, str, None] = None,
+    properties: Optional[dict] = None,
+):
+    """Wrapper around the load_collection, or load_stac method of the openeo connection."""
+    if not is_stac:
+        return connection.load_collection(
+            collection_id=collection_id_or_url,
+            spatial_extent=spatial_extent,
+            temporal_extent=temporal_extent,
+            bands=bands,
+            properties=properties,
+        )
+    cube = connection.load_stac(
+        url=collection_id_or_url,
+        spatial_extent=spatial_extent,
+        temporal_extent=temporal_extent,
+        bands=bands,
+        properties=properties,
+    )
+    print(bands)
+    cube = cube.rename_labels(dimension="bands", target=bands)
```

Comment:
Is there a reason only load_stac needs a rename_labels and not load_collection?

Reply:
There is a problem with the OpenEO python client which doesn't read the bands metadata for PyStac catalogues. It was fixed for the AGERA5 dataset in the `0.30` version but not for the other collections apparently, this is a workaround to keep this metadata.